blocks/market-insight/market-insight.js

```diff
@@ -1,13 +1,16 @@
 import { readBlockConfig, fetchPlaceholders, decorateIcons } from '../../scripts/aem.js';
 import {
-  createElement, observe, postFormData, getResearchAPIUrl, parseResponse,
+  createElement, observe, postFormData, getResearchAPIUrl, parseResponse, SITE_ROOT,
 } from '../../scripts/blocks-utils.js';
 import { handleSocialShareClick } from '../../scripts/social-utils.js';
 
 const apiName = 'GetMarketInsights';
-const apiFormData = '{ pageNo: 1, pageSize: 5 }';
 const defaultCardsCount = 3;
+const MARKETINSIGHT_DIRECTORY = '/research/equity/market-insights/';
 
+function getLink(permLink) {
```

Comment:
every dynamic block would need this kind of method to form the absolute url, would it make sense to have all this at one place and block would just call the required method to get the formed URL.
say `getMarketInsightURL(permLink)` and it returns the absolute url.

Reply:
@ravverma I'll merge this PR, let me know your thoughts on the above then we may make some changes separately.